File: FrontEnd/seeds/campground.js
const mongoose = require("mongoose");
const Review = require("./review");
const Schema = mongoose.Schema;
const Image = require("./images");

const CampgroundSchema = new Schema({
  title: String,

  location: String,
  city: String,
  state: String,
  country: String,
  geometry: {
    type: {
      type: String,
      enum: ["Point"],
      required: true,
    },
    coordinates: {
      type: [Number],
      required: true,
    },
  },
  price: Number,
  description: String,
  author: {
    type: Schema.Types.ObjectId,
    ref: "User",
  },
  reviews: [
    {
      type: Schema.Types.ObjectId,
      ref: "Review",
    },
  ],

  images: [
    {
      type: Schema.Types.ObjectId,
      ref: "Image",
    },
  ],
});

CampgroundSchema.post("findOneAndDelete", async function (doc) {
  if (doc) {
    await Review.deleteMany({
      _id: {
        $in: doc.reviews,
      },
    });
  }
});
CampgroundSchema.post("findOneAndDelete", async function (doc) {
  if (doc) {
    await Image.deleteMany({
      _id: {
        $in: doc.images,
      },
    });
  }
});
module.exports = mongoose.model("Campground", CampgroundSchema);

File: backEnd/src/main/java/com/app/model/Campground.java
package com.app.model;

import java.util.ArrayList;
import java.util.List;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Document("campgrounds")
@NoArgsConstructor
public class Campground {

	@Id
private String id;
	@NotEmpty
private String title;
	@NotEmpty
private String city;
	@NotEmpty
private String state;
	@NotEmpty
private String country;
	@NotNull
private Double price;
	@NotEmpty
private String description;
private String author;
@DBRef private List<Image> images=new ArrayList<>();
@DBRef private List<Review> reviews = new ArrayList<>();

@Override
public String toString() {
	return "Campground [id=" + id + ", title=" + title + ", city=" + city + ", state=" + state + ", country="
			+ country + ", price=" + price + ", description=" + description + ", author=" + author + ", imeges="
			+ images + "]";
}
}
